packages/hebao_v1/contracts/modules/transfers/TransferModule.sol

```diff
@@ -58,38 +58,50 @@ abstract contract TransferModule is SecurityModule
         )
         internal
     {
-        if (amount == 0) return;
-
         if (token == address(0)) {
             transactCall(wallet, to, amount, "");
         } else {
-            bytes memory txData = abi.encodeWithSelector(
-                ERC20(0).transfer.selector,
-                to,
-                amount
+            require(
+                transactTokenTransfer(wallet, token, to, amount),
+                "TRANSFER_FAILED"
             );
-            transactCall(wallet, token, 0, txData);
         }
         emit Transfered(wallet, token, to, amount, logdata);
     }
 
     function approveInternal(
         address wallet,
         address token,
-        address to,
+        address spender,
         uint    amount
         )
         internal
+        returns (uint additionalAllowance)
     {
         require(token != address(0), "UNSUPPORTED");
 
-        bytes memory txData = abi.encodeWithSelector(
-            ERC20(0).approve.selector,
-            to,
-            amount
+        uint allowance = ERC20(token).allowance(wallet, spender);
+        if (allowance >= amount) {
+            return 0;
+        }
```

Comment:
By returning here we disallow lowering the allowance, which we shouldn't always do. Lowering the approval can certainly make sense sometimes (e.g. the spender smart contract was exploited so the approval is removed).

I would remove this check. If the approval doesn't need changing then `approve` should not be called to begin with.

Reply:
I agree